src/select-menu/src/OptionsList.js

```diff
@@ -136,72 +126,60 @@ export default class OptionsList extends PureComponent {
     return fuzzyFilter(options, searchValue, { key: 'label' })
   }
 
-  getCurrentIndex = () => {
-    const { selected } = this.props
-    const options = this.getFilteredOptions()
-
-    return options.findIndex(
-      option => option.value === selected[selected.length - 1]
-    )
-  }
-
-  getFilteredOptions() {
-    const { options } = this.props
-
-    return this.search(options)
-  }
-
   handleKeyDown = e => {
     if (e.keyCode === 38) {
+      // Needed to avoid mixing up navigation on the menu with the filter (if present)
+      e.preventDefault()
       this.handleArrowUp()
     }
 
     if (e.keyCode === 40) {
+      // Needed to avoid mixing up navigation on the menu with the filter (if present)
+      e.preventDefault()
       this.handleArrowDown()
     }
 
     if (e.keyCode === 13) {
+      // Needed to avoid mixing up navigation on the menu with the filter (if present)
+      e.preventDefault()
       this.handleEnter()
     }
   }
 
   handleArrowUp = () => {
-    const { onSelect } = this.props
-    const options = this.getFilteredOptions()
-
-    let nextIndex = this.getCurrentIndex() - 1
-
-    if (nextIndex < 0) {
-      nextIndex = options.length - 1
-    }
-
-    onSelect(options[nextIndex])
```

Comment:
What happened to the `onSelect` callback?

Reply:
Ah, I think we may want to avoid breaking changes in minor/patch versions, to be strictly semver.

Reply:
Yeah. Do we wait for more features before a major version bump? I think that the keyboard navigation is so bogus today that I'd push for this to be the catalyst. This change improves accessibility.

Reply:
Yeah – I think here we'd either merge into a v5 tracking branch instead of `master` if it's a known breaking change.